File: CS/FormatConditionsProvider/Converters/ColorToBrushConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Data;
using System.Windows.Media;
using System.Windows.Markup;

namespace Default_MVVM {
    public class ColorToBrushConverter : MarkupExtension, IValueConverter {
        public ColorToBrushConverter() { }
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            SolidColorBrush brush = value as SolidColorBrush;
            if(brush != null)
                return brush.Color;
            return null;
        }
        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            if(value is Color)
                return new SolidColorBrush((Color)value);
            return null;
        }
        public override object ProvideValue(IServiceProvider serviceProvider) {
            return this;
        }
    }
}

File: VB/FormatConditionsProvider/Converters/ColorToBrushConverter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Data
Imports System.Windows.Media
Imports System.Windows.Markup

Namespace Default_MVVM
	Public Class ColorToBrushConverter
		Inherits MarkupExtension
		Implements IValueConverter
		Public Sub New()
		End Sub
		Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
			Dim brush As SolidColorBrush = TryCast(value, SolidColorBrush)
			If brush IsNot Nothing Then
				Return brush.Color
			End If
			Return Nothing
		End Function
		Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
			If TypeOf value Is Color Then
				Return New SolidColorBrush(CType(value, Color))
			End If
			Return Nothing
		End Function
		Public Overrides Function ProvideValue(ByVal serviceProvider As IServiceProvider) As Object
			Return Me
		End Function
	End Class
End Namespace
